agent/consul/state/intention_test.go

```diff
@@ -1083,6 +1085,124 @@ func TestStore_LegacyIntention_Snapshot_Restore(t *testing.T) {
 	}()
 }
 
+func TestStore_IntentionDecision(t *testing.T) {
+	// web to redis allowed and with permissions
+	// api to redis denied and without perms (so redis has multiple matches as destination)
+	// api to web without permissions and with meta
+	entries := []structs.ConfigEntry{
+		&structs.ProxyConfigEntry{
+			Kind: structs.ProxyDefaults,
+			Name: structs.ProxyConfigGlobal,
+			Config: map[string]interface{}{
+				"protocol": "http",
+			},
+		},
+		&structs.ServiceIntentionsConfigEntry{
+			Kind: structs.ServiceIntentions,
+			Name: "redis",
+			Sources: []*structs.SourceIntention{
+				{
+					Name: "web",
+					Permissions: []*structs.IntentionPermission{
+						{
+							Action: structs.IntentionActionAllow,
+							HTTP: &structs.IntentionHTTPPermission{
+								Methods: []string{"GET"},
+							},
+						},
+					},
+				},
+				{
+					Name:   "api",
+					Action: structs.IntentionActionDeny,
+				},
+			},
+		},
+		&structs.ServiceIntentionsConfigEntry{
+			Kind: structs.ServiceIntentions,
+			Name: "web",
+			Meta: map[string]string{structs.MetaExternalSource: "nomad"},
+			Sources: []*structs.SourceIntention{
+				{
+					Name:   "api",
+					Action: structs.IntentionActionAllow,
+				},
+			},
+		},
+	}
+
+	s := testStateStore(t)
```

Comment:
If you use `testConfigStateStore` here you don't have to drop your own system metadata entry below.